### ISSUE TITLE
Intercepting errors when resolving subscriptions

### ISSUE THREAD

**Issue by LewisYearsley** [mercurius-js/mercurius#966]
Hey. We've been using `errorFormatter` Mercurius option to intercept and log errors. i.e.

```
app.register(mercurius, {
   // ...
    errorFormatter: (executionResult, context) => {
       // Send any errors to our logging service
    },
});
```

This has served us well for queries and mutations, but now we're using subscriptions.

In the case of subscriptions, we've noticed that the callback provided as `errorFormatter` doesn't seem to get run, is that expected?

I've come up with something that works, which is to move all of the logging logic to graphql hooks. i.e.

```
const handleErrorResponse = () => {
  // Do stuff with any errors
}

graphql.addHook("onResolution", handleErrorResponse);
graphql.addHook("onSubscriptionResolution", handleErrorResponse);
```

So to summarise:
- Is it expected that `errorFormatter` isn't executed for subscriptions?
- Is my alternative a good idea? if not, what is the recommended approach, please? 

Using Mercurius 11.5.0

**Comment by mcollina** [comment]
thanks for reporting! The answer is that I don't know, and it's a bit hard to verify without a reproduction script.

This might well be a missing feature in Mercurius. If so, would you like to send a Pull Request?

**Comment by LewisYearsley** [comment]
Thanks, @mcollina - I'll try to figure out what the behaviour should be and based on that I'll be happy to open a PR with the changes if required.

**Comment by LewisYearsley** [comment]
Thanks again @mcollina for getting my changes in - we have them out in prod now and already showing results. We're good to close.
